### ISSUE TITLE
FunctionClauseError: no function clause matching in anonymous fn/2 in XlsxParser.XmlParser.parse_shared_strings/1

### ISSUE THREAD

**Issue by ghost** [TheFirstAvenger/elixir-xlsx_parser#2]
I'm trying to parse a test spreadsheet xlsx file, and I'm getting this error:

``` elixir
** (FunctionClauseError) no function clause matching in anonymous fn/2 in XlsxParser.XmlParser.parse_shared_strings/1
     stacktrace:
       lib/xlsx_parser/xml_parser.ex:50: anonymous fn({:xmlElement, :t, :t, [], {:xmlNamespace, :"http://schemas.openxmlformats.org/spreadsheetml/2006/main", []}, [si: 28, sst: 1], 1, [], [{:xmlText, [t: 1, si: 28, sst: 1], 1, [], 'Text with Newline', :text}, {:xmlText, [t: 1, si: 28, sst: 1], 2, [], '\n', :text}, {:xmlText, [t: 1, si: 28, sst: 1], 3, [], '\n', :text}, {:xmlText, [t: 1, si: 28, sst: 1], 4, [], '\nAnd so on', :text}], [], 'c:/Users/myuser/Projects/myproject', :undeclared}, #HashDict<[{'4', '1.A2'}, {'13', '1.B4'}, {'25', [84, 101, 120, 116, 32, 119, 105, 116, 104, 32, 69, 110, 99, 111, 100, 105, 110, 103, 223, 228, 244, 233]}, {'3', '1.D1'}, {'14', '1.C4'}, {'17', '1.B5'}, {'22', '2.C1'}, {'6', '1.C2'}, {'15', '1.D4'}, {'7', '1.D2'}, {'23', '2.D1'}, {'26', 'Text with Number 123'}, {'16', '1.A5'}, {'2', '1.C1'}, {'18', '1.C5'}, {'11', '1.D3'}, {'19', '1.D5'}, {'0', '1.A1'}, {'9', '1.B3'}, {'8', '1.A3'}, {'21', '2.B1'}, {'24', 'Text'}, {'10', '1.C3'}, {'1', '1.B1'}, {'12', '1.A4'}, {'20', '2.A1'}, {'5', '1.B2'}]>) in XlsxParser.XmlParser.parse_shared_strings/1
       (elixir) lib/enum.ex:1387: Enum."-reduce/3-lists^foldl/2-0-"/3
       lib/xlsx_parser/xlsx_util.ex:12: XlsxParser.XlsxUtil.get_shared_strings/2
       lib/xlsx_parser.ex:20: XlsxParser.get_sheet_content/3
```

Here's the test code:

``` elixir
defmodule XlsxParserTest do
  require Logger
  use ExUnit.Case

  test "parse test spreadsheet" do
    xlsx_file_name = Path.expand("test/fixtures/test_spreadsheet.xlsx")
    expected = [] # TBD
    {:ok, actual} = XlsxParser.get_sheet_content(xlsx_file_name, 1)

    assert actual == expected
  end

end
```

[test_spreadsheet.xlsx](https://github.com/TheFirstAvenger/elixir-xlsx_parser/files/51313/test_spreadsheet.xlsx)

**Comment by TheFirstAvenger** [comment]
So it looks like a newline creates multiple entries in the list being pattern matched here (the one with 'text' as the 5th item in the tuple), as opposed to a single entry we were expecting:

https://github.com/TheFirstAvenger/elixir-xlsx_parser/blob/71ce5886ee09fff74740614cbefc931814783666/lib/xlsx_parser/xml_parser.ex#L50

Will need to modify that pattern match to reduce the list down to a single string. Also need to ensure that double quotes are surrounding the newline or it will break the csv.

Question: Since the newlines are indicated explicitly by the \n characters, what does it mean that there are multiple entries in the list? Should the items in the list be simply concatenated?

**Comment by TheFirstAvenger** [comment]
Fixed error in 08c7e14acbfd523caa97d5fbe8c1d39d2e25d635 (v0.0.7)
